src/lib.rs

```diff
@@ -212,6 +212,17 @@ impl CommitDomain {
         }
     }
 
+    /// Constructs a new `CommitDomain` from different values for `hash_domain` and `blind_domain`
+    pub fn new_with_personalization(hash_domain: &str, blind_domain: &str) -> Self {
+        let m_prefix = format!("{}-M", hash_domain);
```

Comment:
Instead of implementing this by duplication, change `CommitDomain::new` to call this constructor setting both arguments to `domain`.

Reply:
Done